service/discovery/discovery_test.go

```diff
@@ -172,7 +202,193 @@ func TestQuery(t *testing.T) {
 		// If a bad resource was added it will be removed. Otherwise no-op
 		delete(s.resources, "MockResourceId")
 	}
+}
+
+func TestStart(t *testing.T) {
+
+	type envVariable struct {
+		hasEnvVariable   bool
+		envVariableKey   string
+		envVariableValue string
+	}
+
+	type fields struct {
+		hasRPCConnection bool
+		envVariables     []envVariable
+		azLogout         bool
+	}
+
+	tests := []struct {
+		name           string
+		fields         fields
+		wantResp       *discovery.StartDiscoveryResponse
+		wantErr        bool
+		wantErrMessage string
+	}{
+		{
+			name: "Azure authorizer from file",
+			fields: fields{
+				hasRPCConnection: true,
+				envVariables: []envVariable{
+					{
+						hasEnvVariable:   true,
+						envVariableKey:   "AZURE_AUTH_LOCATION",
+						envVariableValue: "service/discovery/testfile/credentials_test_file",
```

Comment:
Do we already have a folder for test files somewhere? Or maybe rename the directory `testfile` to testing?